crates/scroll/node/src/builder/network.rs

```diff
@@ -55,9 +111,36 @@ where
         ctx: &BuilderContext<Node>,
         pool: Pool,
     ) -> eyre::Result<Self::Network> {
+        // initialize the rollup node database.
+        let db_path = ctx.config().datadir().db();
+        let database_path = if let Some(database_path) = self.rollup_node_db_path {
+            database_path.to_string_lossy().to_string()
+        } else {
+            // append the path using strings as using `join(...)` overwrites "sqlite://"
+            // if the path is absolute.
+            let path = db_path.join("scroll.db?mode=rwc");
+            "sqlite://".to_string() + &*path.to_string_lossy()
+        };
+        let db = Arc::new(Database::new(&database_path).await?);
+
         // get the header transform.
         let chain_spec = ctx.chain_spec();
-        let transform = ScrollHeaderTransform { chain_spec };
+        let authorized_signer = if self.signer.is_none() {
+            match chain_spec.chain().named() {
+                Some(NamedChain::Scroll) => Some(SCROLL_MAINNET_SIGNER),
+                Some(NamedChain::ScrollSepolia) => Some(SCROLL_SEPOLIA_SIGNER),
+                _ => None,
```

Comment:
Why do we do this? Can't we leave it as the responsiblity of the client to corectly set the autorize signer?

Reply:
Sorry, I'm mistaken. I think this is the correct place to determine the signer. But we should move these modifications into the `ScrollNetworkBuilder ` implementation in  `rollup-node`